src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -472,19 +530,27 @@ public function dumpFile($filename, $content, $mode = 0666)
             throw new IOException(sprintf('Unable to write to the "%s" directory.', $dir), 0, null, $dir);
         }
 
-        $tmpFile = tempnam($dir, basename($filename));
+        $tmpFile = $this->tempNam($dir, basename($filename));
 
         if (false === @file_put_contents($tmpFile, $content)) {
             throw new IOException(sprintf('Failed to write file "%s".', $filename), 0, null, $filename);
         }
 
         $this->rename($tmpFile, $filename, true);
+
         if (null !== $mode) {
+            $scheme = $this->getScheme($filename);
+            if (false !==  $scheme && 'file' !== $scheme) {
+                trigger_error('Support for modifying file permissions on streams is not implemented.', E_USER_WARNING);
```

Comment:
On this part of the patch, I'd revert it: it's a deprecated feature, we don't need to make it work for the new use cases. The will discourage/prevent new code being written that takes advantage of your bug fix and also uses the deprecated $mode arg.